module/zfs/zio.c

```diff
@@ -1889,8 +1961,16 @@ zio_wait(zio_t *zio)
 	__zio_execute(zio);
 
 	mutex_enter(&zio->io_lock);
-	while (zio->io_executor != NULL)
-		cv_wait_io(&zio->io_cv, &zio->io_lock);
+	while (zio->io_executor != NULL) {
+		error = cv_timedwait_io(&zio->io_cv, &zio->io_lock,
+		    ddi_get_lbolt() + MSEC_TO_TICK(zfs_deadman_checktime_ms));
+
+		if (error == -1) {
+			uint64_t delta = gethrtime() - zio->io_queued_timestamp;
+			if (delta > spa_deadman_ziotime(zio->io_spa))
```

Comment:
We probably want to add an additional check here such that `zio_deadman()` can only run once for a hung `zio`.  Or alternately it runs at the lower frequency of `zfs_deadman_ziotime_ms` intervals, instead of every `zfs_deadman_checktime_ms` which is what the code does now.

Reply:
I opted to leave this as-is such that it's consistent with the `zfs_deadman_synctime_ms` behavior.  The document ion has been updated accordingly.